### PR TITLE
feat: invite user by email

### PR DESCRIPTION
**Pull request by dplynskiy**
- new user type - invited user
- new button - send invite - sends link to signup page for invited user, to login page for all other
- new fields at email provider - email Invite email title and Invite email content
- invited user can't login or reset password
- when invited user signup - user fields updated with signup form data, type is changing to normal-user

### CODE REVIEW

File: controllers/user.go [hunk 1]
```diff
@@ -577,3 +583,83 @@ func (c *ApiController) RemoveUserFromGroup() {
...
+		return
+	}
+
+	sender := organization.DisplayName
+	origin := conf.GetConfigString("origin")
+
+	var link string
+	switch user.Type {
+	case "invited-user":
+		link = fmt.Sprintf("%s/signup/%s?id=%s&u=%s&e=%s", origin, application.Name, user.Id, user.Name, user.Email)
```
Comment (ISulimanov) [inline controllers/user.go:640]: does user.Email has mask?

**Review by ISulimanov — CHANGES_REQUESTED** [review]

File: controllers/user.go (hunk 1 above)
Comment (dplynskiy) [reply controllers/user.go:640]: no, masked email require change logic at send verification code method. Is it critical?

File: controllers/auth.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func (c *ApiController) HandleLoggedIn(application *object.Application, user *ob
 		c.ResponseError(err.Error(), nil)
 		return
 	}
-	if !allowed {
+	if !allowed || user.Type == "invited-user" {
```
[the remark this answers is not in the document]
Comment (dplynskiy) [reply controllers/auth.go]: done

File: controllers/account.go [hunk 3]
```diff
@@ -194,10 +194,67 @@ func (c *ApiController) Signup() {
 		}
 	}
 
-	affected, err := object.AddUser(user)
-	if err != nil {
-		c.ResponseError(err.Error())
-		return
+	var affected bool
+
+	if authForm.Id != "" {
```
[the remark this answers is not in the document]
Comment (dplynskiy) [reply controllers/account.go]: done

**Review by ISulimanov — APPROVED** [review]

**Pull request merged**